### ISSUE TITLE
[Accessibility] Improve support for Windows High Contrast Mode.

### ISSUE THREAD

**Issue by majornista** [adobe/spectrum-css#168]
Windows High Contrast Mode provides users with a selection of different ready-made high-contrast themes.

How to test: https://support.microsoft.com/en-us/help/4026951/windows-turn-high-contrast-mode-on-or-off-in-windows

Different options here would be High Contrast Black, High Contrast white, etc. Each theme maps the font and background colors in the application to system colors. Like below.

![image](https://user-images.githubusercontent.com/154077/57879916-7f673180-77eb-11e9-851b-2aedfd5444d0.png)

Regardless of which specific color scheme a user has chosen, -ms-high-contrast:active media query will evaluate to true when any of these high contrast themes are enabled. Adding these styles ensure that users can identify the components in different states properly. Am using the CSS2 system colors that are mapped to Windows High Contrast colors. Like highlight, highlightText, GrayText, windowText and outline properties to convey the different states of the component. And for a few components borders had to be added because relying on background colors alone to represent state changes won’t work in high contrast mode (Because the browser strips off background colors in high contrast mode)

These are also minimal set of changes required to make the high contrast themes render components correctly.

Providing high contrast mode options for all the browsers is tricky because each browser has very widely varying support. This PR fixes the high contrast mode problem for browser combination with best support (Windows/ IE & Edge).

More reading: http://www.gwhitworth.com/blog/2017/04/how-to-use-ms-high-contrast

https://github.com/MicrosoftEdge/MSEdgeExplainers/blob/master/Accessibility/HighContrast/explainer.md

**Comment by Westbrook** [comment]
Is there any motion on getting this prioritized @lazd @misterbrownlee?

**Comment by NateBaldwinDesign** [comment]
Spectrum design is currently investigating what we would expect these components to look like in windows high contrast mode.

**Comment by GarthDB** [comment]
@majornista Work is happening. @jnurthen submitted a pr for Checkbox. Should we close this in favor of individual issues? I wouldn't mind making a milestone for it as well.

**Comment by misterbrownlee** [comment]
We're looking into adding testing cross browser, which will improve things like this getting overlooked.  We also have added WHCM in general via help from James N.  Closing this general request, if there are specific components that need WHCM support or improvement let's open a more focused ticket

**Comment by jnurthen** [comment]
I still have a lot more to forward port from react-spectrum. Am on vacation this week but on my list when I get back.
